### PR TITLE
Add authorization

### PR DESCRIPTION
**Pull request by d2ndjim**
- Added CanCanCan to project
- Added role column to users table with migration
- Used CanCanCan to authorize A user to delete a post if it is theirs or if they have an admin role.
- Used CanCanCan to authorize A user to delete a comment if it is theirs or if they have an admin role

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -12,4 +12,8 @@ class User < ApplicationRecord
   def three_recent_posts
     Post.last(3)
   end
+
+  def admin?
+    :role == 'admin'
+  end
```
Comment (TedLivist) [inline app/models/user.rb]: - [ ] This function does not seem to work as expected. I created an admin on the console and signed it in on the browser, but when I visit a post I don't see the button for deleting a post or a comment as the admin.

  You can resolve this by refactoring the function to the code below, so we can actually have access to the User object that calls the method and its `role` attribute.

  ```
  def admin?
    self.role == 'admin'
  end
  ```  

  adm

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @d2ndjim and @Yazino12,

Good job so far!
✔️ Good documentation
✔️ Implementation of authorization using cancancan
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by d2ndjim** [comment]
Hello @TedLivist, The delete button is posed to be displayed on the post shows page, the image below is a screenshot of the app on my computer

image

**Review by descholar-ceo — APPROVED** [review]
Hi @d2ndjim && @Yazino12 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Your feature works fine ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**